src/main/java/com/minecolonies/core/entity/ai/workers/AbstractEntityAIStructure.java

```diff
@@ -477,6 +484,10 @@ else if (result.getBlockResult().getResult() == BlockPlacementResult.Result.LIMI
             this.storeProgressPos(result.getIteratorPos(), structurePlacer.getB().getStage());
         }
 
+        if (firstIteration)
+        {
+            building.checkOrRequestBucket(building.getRequiredResources(), worker.getCitizenData());
```

Comment:
isnt the building progress saved to nbt? may lead to problems I guess

Reply:
Yes, that's why this should only ever be called once per building